modules/apps/29-fee/keeper/msg_server.go

```diff
@@ -1,4 +1,33 @@
 package keeper
 
-// TODO
-//var _ types.MsgServer = Keeper{}
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/modules/apps/29-fee/types"
+)
+
+var _ types.MsgServer = Keeper{}
+
+// RegisterCounterpartyAddress is called by the relayer on each channelEnd and allows them to specify their counterparty address before relaying
+// This ensures they will be properly compensated for forward relaying since destination chain must send back relayer's source address (counterparty address) in acknowledgement
+// This function may be called more than once by relayer, in which case, latest counterparty address is always used.
```

Comment:
@AdityaSripal I took this to mean that the address can be overwritten by calling `RegisterCounterPartyAddress` more than once. Is that correct?

Reply:
Correct